dask/dataframe/tests/test_multi.py

```diff
@@ -262,6 +264,168 @@ def test_sequential_joins():
     assert_eq(multi_join_pd, multi_join_dd)
 
 
+def test_merge_asof_index():
+    A = pd.DataFrame({'left_val': list('abcd' * 3)}, index=[1,3,7,9,10,13,14,17,20,24,25,28])
+    a = dd.from_pandas(A, npartitions=4)
+    B = pd.DataFrame({'right_val': list('xyz' * 4)}, index=[1,2,3,6,7,10,12,14,16,19,23,26])
+    b = dd.from_pandas(B, npartitions=3)
+
+    C = pd.merge_asof(A, B, left_index=True, right_index=True)
+    c = merge_asof(a, b, left_index=True, right_index=True)
+
+    assert_eq(c, C)
+
+
+def test_merge_asof_on():
+    A = pd.DataFrame({'a': [1, 5, 10], 'left_val': ['a', 'b', 'c']})
+    a = dd.from_pandas(A, npartitions=2)
+    B = pd.DataFrame({'a': [1, 2, 3, 6, 7], 'right_val': [1, 2, 3, 6, 7]})
+    b = dd.from_pandas(B, npartitions=2)
+
+    C = pd.merge_asof(A, B, on='a').set_index('a')
+    c = merge_asof(a, b, on='a')
+    assert_eq(c, C)
+
+
+def test_merge_asof_on_no_exact_matches():
+    A = pd.DataFrame({'a': [1, 5, 10], 'left_val': ['a', 'b', 'c']})
+    a = dd.from_pandas(A, npartitions=2)
+    B = pd.DataFrame({'a': [1, 2, 3, 6, 7], 'right_val': [1, 2, 3, 6, 7]})
+    b = dd.from_pandas(B, npartitions=2)
+
+    C = pd.merge_asof(A, B, on='a', allow_exact_matches=False).set_index('a')
```

Comment:
Why is the `set-index` needed here? I haven't looked at the implementation yet, forgive me if that's explained there.

Reply:
Sorry. When Pandas does a merge_asof on a particular column, the resulting dataframe gets reindexed 0, 1, .... This is the one place where my implementation differs from Pandas': if your merge_asof on a particular column using mine, it sets the index as that on column. I could alternatively do a reset_index() in my implementation but I find it kind of silly.

Reply:
Furthermore, this is causing Travis CI to mess up, since I guess this causes some index to be permuted in a different way.

```
E       AssertionError: Index are different
E       
E       Index values are different (100.0 %)
E       [left]:  Index([u'price', u'quantity', u'ticker', u'ask', u'bid'], dtype='object')
E       [right]: Index([u'ask', u'bid', u'price', u'quantity', u'ticker'], dtype='object')
```

Do you know what I should do to fix this to pass Travis CI?

Reply:
Ideally we would match pandas output.

We may not be able to entirely in this case. This might be a case where we need each partition to have its own `RangeIndex(0, len(partition))`.